src/mm/checkpoints.c

```diff
@@ -499,10 +628,9 @@ void restore_full(int lid, void *ckpt) {
 void log_restore(int lid, state_t *state_queue_node) {
 	statistics_post_lp_data(lid, STAT_RECOVERY, 1.0);
 	
-	if(pdes_config.checkpointing == INCREMENTAL_STATE_SAVING){
+	if(pdes_config.iss_enabled_mprotection == MPROTECT && pdes_config.checkpointing == INCREMENTAL_STATE_SAVING){
 		iss_unprotect_all_memory(lid);
 		restore_full(lid, state_queue_node->log);
-		iss_protect_all_memory(lid);
```

Comment:
at least one should be left to protect again the segment

Reply:
added protect all memory again